batch/batch/cloud/gcp/resource_utils.py

```diff
@@ -11,21 +11,37 @@
 
 MACHINE_FAMILY_TO_ACCELERATOR_VERSIONS = {'g2': 'l4'}
 
+SINGLE_GPU_MACHINES = ['g2-standard-4', 'g2-standard-8', 'g2-standard-12', 'g2-standard-16', 'g2-standard-32']
+
+TWO_GPU_MACHINES = [
+    'g2-standard-24',
+]
+
+FOUR_GPU_MACHINES = [
+    'g2-standard-48',
+]
+
+EIGHT_GPU_MACHINES = [
+    'g2-standard-96',
+]
```

Comment:
nit: Instead of these different lists you could just have a dictionary of `machine_type` to number of GPUs. That also reduces the logic in the prior function